fdk.go

```diff
@@ -115,26 +115,25 @@ func doJSON(handler Handler, ctx context.Context, in io.Reader, out io.Writer) {
 	var buf bytes.Buffer
 	hdr := make(http.Header)
 
+	resp := &jsonOut{}
+	req := &jsonIn{}
+
 	for {
-		doJSONOnce(handler, ctx, in, out, &buf, hdr)
+		doJSONOnce(handler, ctx, in, out, resp, req, &buf, hdr)
```

Comment:
not sure this is that useful, this function signature is really bloated now. buffer and headers both have sizable maps allocated which are useful to re-use. this is just a pointer. in scope what I meant by using var syntax was:

```go
var resp jsonOut
// ...
doAThing(&resp)
```

Reply:
this has bad side effects as well (I can elaborate, if desired)

Reply:
fixed.